packages/monaco/src/browser/monaco-keybinding-contexts.ts

```diff
@@ -41,3 +44,32 @@ export class MonacoStrictEditorTextFocusContext extends StrictEditorTextFocusCon
     }
 
 }
+
+/**
+ * The Monaco editor itself is a `textArea` so we have to restrict the default native text input focus context.
+ * It is enabled if the focus is on an `input` or `textArea` and not contained in a Monaco editor.
+ */
+@injectable()
+export class MonacoNativeTextInputFocusContext extends NativeTextInputFocusContext {
```

Comment:
I wonder can we get aways without it? As far as I understood monaco handler now will be evaluated first anyway?

Reply:
I need to check; if possible, sure let's remove it.

Reply:
I cannot remove this. If we do not use this customization we have two matching bindings here
https://github.com/eclipse-theia/theia/blob/cf86e86857c45ab82366508cd270d217d6d66b04/packages/core/src/browser/keybinding.ts#L615

These are the two valid bindings:

![Screen Shot 2020-04-02 at 09 14 45](https://user-images.githubusercontent.com/1405703/78220891-a80b1f00-74c2-11ea-98d2-51cde30fab0c.jpg)


And we blindly pick the first one, which is the core and it does not work.

Reply:
ok, probably noone will try to rebind `NativeTextInputFocusContext`, so it should be fine

Reply:
I added the above-described caveat as a jsdoc.